src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.cs

```diff
@@ -374,7 +374,15 @@ protected SynthesizedImplementationMethod GenerateIteratorGetEnumerator(MethodSy
             //    {
             //        result = new {StateMachineType}({initialState});
             //    }
-            //    result.parameter = this.parameterProxy; // copy all of the parameter proxies
+            //
+            //    // copy each parameter proxy
+            //    result.parameter = this.parameterProxy;
```

Comment:
I'm not understanding this. This line and the line in the true `if` block below are the same, what's the purpose of that?